agent.go

```diff
@@ -303,10 +303,18 @@ func (a *Agent) flush(bp *BatchPoints) error {
 
 	for _, o := range a.outputs {
 		wg.Add(1)
+
+		// Copy BatchPoints
+		var bpc BatchPoints
+		bpc.Time = bp.Time
+		bpc.Tags = bp.Tags
+		bpc.Precision = bp.Precision
+		copy(bp.Points, bpc.Points)
```

Comment:
I think your args might be reversed.  This looks like it will copy the nil `bpc.Points` into `bp.Points`.